File: Base Dati/geometria/triangolo.py
from punto import Punto
from segmento import Segmento
from math import sqrt

class Triangolo:
    def __init__(self, A:Punto, B:Punto, C:Punto) -> None:
        self.A = A
        self.B = B
        self.C = C

        self.AB = Segmento(A,B)
        self.AC = Segmento(A,C)
        self.BC = Segmento(B,C)

    def perimetro(self):
        return self.AB.lunghezza() + self.AC.lunghezza() + self.BC.lunghezza()
    
    def area(self):
        sp = self.perimetro() / 2
        return sqrt(  sp * (sp -self.AB.lunghezza()) * (sp -self.AC.lunghezza()) * (sp -self.BC.lunghezza())  )

    def __str__(self) -> str:
        return f"Triangolo con perimentro di {self.perimetro()} e superficie di {self.area()}"

File: Java/Prj05_GeometriaPiana/src/figure/Triangolo.java
package figure;

public class Triangolo  extends LuogoGeometrico implements Contornabile{

    Punto a,b,c;
    Segmento ab, ac, bc;

    public Triangolo(Punto a, Punto b, Punto c) {
        this.nome = "Triangolo";
        this.a = a;
        this.b = b;
        this.c = c;
        this.ab = new Segmento(a,b);
        this.ac = new Segmento(a,c);
        this.bc = new Segmento(b,c);
    }

    public double perimetro(){
        return ab.lunghezza() + ac.lunghezza() + bc.lunghezza();
    }

    public double area(){
        double sp = perimetro() / 2;
        return Math.sqrt(
                sp *
                (sp-ab.lunghezza()) *
                (sp-ac.lunghezza()) *
                (sp-bc.lunghezza())
        );
    }

    @Override
    public double contorno() {
        return perimetro();
    }

    @Override
    public String toString() {
        return "Triangolo[" +
                "Perimetro=" + perimetro() +
                ", Area=" + area() +
                ']';
    }
}
